### PR TITLE
Update documentation in `pillarbox-ui`

### PR DESCRIPTION
**Pull request by MGaetan89**
# Pull request

## Description

Update the documentation in `pillarbox-ui`.

## Changes made

- Update `docs/README.md` to work with Dokka (mainly removing images) and improving wording and samples.
- Remove `Module.md`.
- Update the KDoc for every public class/method/interface.

## Checklist

- [x] APIs have been properly documented (if relevant).
- [x] The documentation has been updated (if relevant).
- [ ] New unit tests have been written (if relevant).
- [ ] The demo has been updated (if relevant).

### CODE REVIEW

File: pillarbox-ui/src/main/java/ch/srgssr/pillarbox/ui/widget/DelayedVisibilityState.kt [hunk 1]
```diff
@@ -107,23 +114,23 @@ class DelayedVisibilityState internal constructor(
...
 }
 
 /**
- * Toggleable
+ * Makes a Composable toggleable, controlling the visibility state of a [DelayedVisibilityState].
  *
- * @param enabled whether this toggleable will handle input events and appear enabled for semantics purposes
- * @param role the type of user interface element. Accessibility services might use this to describe the element or do customizations
- * @param delayedVisibilityState the delayed visibility state to link
+ * @param enabled Whether to handle input events and appear enabled for semantics purposes.
```
Comment (defagos) [inline pillarbox-ui/src/main/java/ch/srgssr/pillarbox/ui/widget/DelayedVisibilityState.kt:131]: Note sure _semantic_ suits well here. Is there a purpose related to a meaning here (really what I understand by _semantic_) or are you rather talking about some kind of overall behavior perceived externally?

File: pillarbox-ui/src/main/java/ch/srgssr/pillarbox/ui/widget/DelayedVisibilityState.kt [hunk 2]
```diff
@@ -137,14 +144,13 @@ fun Modifier.toggleable(
...
- * @param enabled whether this toggleable will handle input events and appear enabled for semantics purposes
- * @param role the type of user interface element. Accessibility services might use this to describe the element or do customizations
- * @param indication indication to be shown when the modified element is pressed. By default, indication from LocalIndication will be used.
- * Pass null to show no indication, or current value from LocalIndication to show theme default
- * @param interactionSource MutableInteractionSource that will be used to emit PressInteraction.Press when this toggleable is being pressed.
- * @param delayedVisibilityState the delayed visibility state to link
+ * @param enabled Whether to handle input events and appear enabled for semantics purposes.
+ * @param role The type of UI element. Accessibility services might use this to describe the element.
+ * @param indication Indication to be shown when the Composable is pressed. If `null`, no indication will be shown.
+ * @param interactionSource The [MutableInteractionSource] that will be used to dispatch [Interaction]s when this toggleable is being interacted with.
```
Comment (defagos) [inline pillarbox-ui/src/main/java/ch/srgssr/pillarbox/ui/widget/DelayedVisibilityState.kt:152]: Maybe a word is missing here? 

> toggleable _component_

**Review by defagos — APPROVED** [review]

File: pillarbox-ui/src/main/java/ch/srgssr/pillarbox/ui/widget/DelayedVisibilityState.kt (hunk 1 above)
Comment (MGaetan89) [reply pillarbox-ui/src/main/java/ch/srgssr/pillarbox/ui/widget/DelayedVisibilityState.kt:131]: I've reused the wording used in Compose's `toggleable` `Modifier`: https://developer.android.com/reference/kotlin/androidx/compose/foundation/selection/package-summary#(androidx.compose.ui.Modifier).toggleable(kotlin.Boolean,kotlin.Boolean,androidx.compose.ui.semantics.Role,kotlin.Function1)
> whether or not this [toggleable](https://developer.android.com/reference/kotlin/androidx/compose/foundation/selection/package-summary#(androidx.compose.ui.Modifier).toggleable(kotlin.Boolean,kotlin.Boolean,androidx.compose.ui.semantics.Role,kotlin.Function1)) will handle input events and appear enabled for semantics purposes

It refers to the accessibility part of the component, ie. talkback (the screen reader) will indicate that the component is disabled.

File: pillarbox-ui/src/main/java/ch/srgssr/pillarbox/ui/widget/DelayedVisibilityState.kt (hunk 2 above)
Comment (MGaetan89) [reply pillarbox-ui/src/main/java/ch/srgssr/pillarbox/ui/widget/DelayedVisibilityState.kt]: 🫣

**Review by defagos — APPROVED** [review]

**Pull request merged**